Sources/SwiftInspectorCommands/PropertyAnalyzerCommand.swift

```diff
@@ -0,0 +1,76 @@
+// Created by Tyler Hedrick on 8/17/20.
+//
+// Copyright (c) 2020 Tyler Hedrick
+//
+// Distributed under the MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import ArgumentParser
+import Foundation
+import SwiftInspectorCore
+
+final class PropertyAnalyzerCommand: ParsableCommand {
+  static var configuration = CommandConfiguration(
+    commandName: "properties",
+    abstract: "Finds property information for the provided type"
+  )
+
+  @Option(help: nameArgumentHelp)
+  var name: String
+
+  @Option(help: "The absolute path of the file to inspect")
+  var path: String
+
+  /// Runs the command
+  func run() throws {
+    let cachedSyntaxTree = CachedSyntaxTree()
+
+    let analyzer = PropertyAnalyzer(typeName: name, cachedSyntaxTree: cachedSyntaxTree)
+    let fileURL = URL(fileURLWithPath: path)
+    if let propertyInformation = try analyzer.analyze(fileURL: fileURL) {
+      print(outputString(from: propertyInformation))
+    }
+  }
+
+  /// Validates if the arguments of this command are valid
+  func validate() throws {
+    guard !name.isEmpty else {
+      throw InspectorError.emptyArgument(argumentName: "--name")
+    }
+    guard !path.isEmpty else {
+      throw InspectorError.emptyArgument(argumentName: "--path")
+    }
+    let pathURL = URL(fileURLWithPath: path)
+    guard FileManager.default.isSwiftFile(at: pathURL) else {
+      throw InspectorError.invalidArgument(argumentName: "--path", value: path)
+    }
+  }
+
+  private func outputString(from statement: TypeWithPropInfo) -> String {
+    let propString = statement.properties.map { propInfo in
+      "(\(propInfo.name)|\(propInfo.access)|\(propInfo.scope))"
```

Comment:
Should we use spaces or commas to separate these and then new lines for each property instead? this would be the first command that uses`|` to separate fields

Reply:
Yeah, i wasn't sure what to do here exactly, im good with changing it to what you said 👍

Reply:
So let's say that you ran this command like `./SwiftInspector properties -name Some -path /whatever.swift` and the type looked like this:

```swift
class Some {
  private let hello: Int
  public let goodbye: String
}
```

I think output like the following would look be good.

```
hello, private, instance
goodbye, public, instance
```

I don't think we need to regurgitate the type name since you passed it explicitly into the command.

(Apologies if I don't _fully_ understand the output but I _think_ this close and should give an idea of where my head is at 👍 )

Reply:
I think returning the type is useful when parsing the result. That way you don't need to inspect both what you're passing and what the output is giving you. I've found it useful at least

Reply:
I think it probably depends on whether you are invoking this command from another script (e.g. Ruby) or running it manually on the command line.

Reply:
@fdiaz should I add the type information back into the output?